src/etc/inc/gwlb.inc

```diff
@@ -864,6 +869,15 @@ function return_gateways_array($disabled = false, $localhost = false, $inactive
 		$gateways_arr['Null4'] = $gwlo4;
 		$gateways_arr['Null6'] = $gwlo6;
 	}
+
+	if ($integer_index) {
+		$gateways_arr_temp = array();
```

Comment:
Yes, I looked at that and thought so. But then I got worried about "was there some reason it was coded like this?" and so I just refactored-out the repeated code into the function, so that at least it only appeared once. (it wasn't my objective at the time to "fixup" this)
The subsequent use of $gateways_arr assumes that the index into $gateways_arr of gateways that do exist in the config is the same index they have in the config. The way that return_gateways_array() builds its list means that the array entries should be added in the same order as in the config.
That order must not be messed up by re-indexing the keys.
I will do a few tests/checks to make sure that array_values() will not cause a problem.